lib/ajaxful_rating/stars_icon_builder.rb

```diff
@@ -0,0 +1,117 @@
+module AjaxfulRating # :nodoc:
+  class StarsIconBuilder # :nodoc:
+    include AjaxfulRating::Locale
+
+    attr_reader :rateable, :user, :options, :remote_options, :show_value
+
+    def initialize(rateable, user_or_static, template, options = {}, remote_options = {})
+      @user = user_or_static unless user_or_static == :static
+      @rateable = rateable
+      @template = template
+      apply_stars_builder_options!(options, remote_options)
+      @show_value = calculate_show_value
+    end
+
+    def calculate_show_value
+      if options[:show_user_rating]
+        rate = rateable.rate_by(user, options[:dimension]) if user
+        rate ? rate.stars : 0
+      else
+        rateable.rate_average(true, options[:dimension])
+      end
+    end
+
+    def render
+      options[:wrap] ? wrapper_tag : ratings_tag
+    end
+
+    private
+      def apply_stars_builder_options!(options, remote_options)
+        @options = {
+          wrap: true,
+          show_user_rating: false,
+          force_static: false,
+          current_user: (@template.current_user if @template.respond_to?(:current_user))
+        }.merge(options)
+
+        # Do not see the need at the moment
```

Comment:
Where does this come from? Can we maybe just delete those lines?

Reply:
This is legacy code that I didnt see the point of.